rayforge/mainwindow.py

```diff
@@ -1205,12 +1205,156 @@ def on_copy_requested(self, sender, items: List[DocItem]):
     def on_paste_requested(self, sender, *args):
         """
         Handles the 'paste-requested' signal from the WorkSurface.
+        Checks for image data on system clipboard first, then falls back
+        to workpiece paste. After pasting an image, the clipboard is
+        cleared so subsequent pastes use workpiece data.
         """
+        # Priority 1: Check if system clipboard contains image data
+        clipboard = self.get_clipboard()
+        formats = clipboard.get_formats()
+        has_image = formats.contain_mime_type("image/png") or \
```

Comment:
Either accept any support format by checking the existing `image.importer_by_mime_type`, or define a `mime_formats_that_support_drag_drop` map in the image module, if there is a good reason to limit this